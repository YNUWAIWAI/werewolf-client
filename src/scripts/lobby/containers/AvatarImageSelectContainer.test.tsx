import * as ActionTypes from '../constants/ActionTypes'
import * as React from 'react'
import {AvatarImageList} from '../constants/AvatarImageList'
import {AvatarImageSelect} from '../components/organisms/AvatarImageSelect'
import {AvatarImageSelectContainer} from './AvatarImageSelectContainer'
import {ImagePath} from '../constants/ImagePath'
import IntlProviderContainer from './IntlProviderContainer'
import {Provider} from 'react-redux'
import fakeStore from './fakeStore'
import {mount} from 'enzyme'

test('render', () => {
  const token = '3F2504E0-4F89-11D3-9A0C-0305E82C3300'
  const store = fakeStore({
    avatarImageList: {
      imageList: AvatarImageList,
      scope: ActionTypes.Scope.SelectHumanAvatar,
      selectedImage: ImagePath.Character.a,
      token
    }
  })
  const wrapper = mount(
    <Provider store={store} >
      <IntlProviderContainer>
        <AvatarImageSelectContainer />
      </IntlProviderContainer>
    </Provider>
  )

  expect(wrapper.html()).toMatchSnapshot()
})
test('handleCloseButtonClick', () => {
  const token = '3F2504E0-4F89-11D3-9A0C-0305E82C3300'
  const store = fakeStore({
    avatarImageList: {
      imageList: AvatarImageList,
      scope: ActionTypes.Scope.SelectHumanAvatar,
      selectedImage: ImagePath.Character.a,
      token
    }
  })
  const dispatch = jest.fn()

  store.dispatch = dispatch
  const wrapper = mount(
    <Provider store={store} >
      <IntlProviderContainer>
        <AvatarImageSelectContainer />
      </IntlProviderContainer>
    </Provider>
  )

  wrapper.find(AvatarImageSelect).props().handleCloseButtonClick()
  expect(dispatch).toHaveBeenCalledTimes(1)
  expect(dispatch).toHaveBeenCalledWith({
    type: ActionTypes.AvatarImageSelect.CLOSE_MODAL
  })
})
test('handleImageSelect', () => {
  const token = '3F2504E0-4F89-11D3-9A0C-0305E82C3300'
  const store = fakeStore({
    avatarImageList: {
      imageList: AvatarImageList,
      scope: ActionTypes.Scope.SelectHumanAvatar,
      selectedImage: ImagePath.Character.a,
      token
    }
  })
  const dispatch = jest.fn()

  store.dispatch = dispatch
  const wrapper = mount(
    <Provider store={store} >
      <IntlProviderContainer>
        <AvatarImageSelectContainer />
      </IntlProviderContainer>
    </Provider>
  )
  const image = ImagePath.Character.b

  wrapper.find(AvatarImageSelect).props().handleImageSelect(image)
  expect(dispatch).toHaveBeenCalledTimes(1)
  expect(dispatch).toHaveBeenCalledWith({
    image,
    type: ActionTypes.AvatarImageSelect.SELECT_AVATAR
  })
})
