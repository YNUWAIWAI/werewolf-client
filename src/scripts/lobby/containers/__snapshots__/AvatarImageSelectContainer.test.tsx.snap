// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarImageSelectContainer /> render 1`] = `"<div class=\\"lo--avatar-image-select\\"><button class=\\"lo--avatar-image-select--close\\"><svg class=\\" close\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"><path d=\\"M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2C6.47,2 2,6.47 2,12C2,17.53 6.47,22 12,22C17.53,22 22,17.53 22,12C22,6.47 17.53,2 12,2M14.59,8L12,10.59L9.41,8L8,9.41L10.59,12L8,14.59L9.41,16L12,13.41L14.59,16L16,14.59L13.41,12L16,9.41L14.59,8Z\\"></path></svg></button><div class=\\"lo--avatar-image-select--description\\">Choose an image</div><div class=\\"lo--avatar-image-select--image-list\\"><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image selected\\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/a_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/b_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/c_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/d_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/e_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/f_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/g_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/h_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/i_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/j_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/k_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/l_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/m_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/n_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/o_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/r_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/s_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/t_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/u_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/v_120x120.png\\"></div><div class=\\"lo--avatar-image-select--image-list--item\\"><img class=\\"lo--avatar-image-select--image-list--item--image \\" src=\\"https://werewolf.world/image/0.3/character_icons/120x120/y_120x120.png\\"></div></div></div>"`;
